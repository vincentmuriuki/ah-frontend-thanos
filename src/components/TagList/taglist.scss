.tag{
  background-color:#efefef;
  text-align: center;
  text-transform: capitalize;
  font-size: 15px ! important;
}
.tag:hover{
  background: #20c0b0;
  color: #ffffff;
  cursor: pointer;
}
.badge-pill {
 margin: 4px 40px;
}