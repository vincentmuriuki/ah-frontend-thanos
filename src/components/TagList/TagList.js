import React from 'react';
import PropTypes from 'prop-types';
import './taglist.scss';

const TagList = ({ tags }) => tags.map((tag, id) => (
  <div key={id} className="badge-pill h4 px-3 tag">
    {tag.name}
  </div>
));

TagList.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.object),
};
TagList.defaultProps = {
  tags: [{}],
};

export default TagList;
