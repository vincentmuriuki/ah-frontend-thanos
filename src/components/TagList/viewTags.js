import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getTag } from '../../actions/tagsActions';
import TagList from './TagList';


export class Tags extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(getTag());
  }

  render() {
    const { tags } = this.props;
    return (<TagList tags={tags} key={tags.id} />);
  }
}

Tags.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.object),
  dispatch: PropTypes.func.isRequired,
};
Tags.defaultProps = {
  tags: [{}],
};
export const mapStateToProps = ({ tagsReducer }) => ({ tags: tagsReducer.tags });
export default connect(mapStateToProps)(Tags);
