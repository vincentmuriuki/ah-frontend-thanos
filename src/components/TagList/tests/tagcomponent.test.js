import React from 'react';
import { shallow } from 'enzyme';
import TagList from '../TagList';

it('should render correctly', () => {
  const wrapper = shallow(
    <TagList />,
  );
  expect(wrapper).toMatchSnapshot();
});
