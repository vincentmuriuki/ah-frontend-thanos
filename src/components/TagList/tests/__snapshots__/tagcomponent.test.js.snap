// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="badge-pill h4 px-3 tag"
  key="0"
/>
`;
